fairmq/shmem/Message.h

```diff
@@ -265,6 +263,7 @@ class Message final : public fair::mq::Message
                 if (fManager.ThrowingOnBadAlloc()) {
                     throw MessageBadAlloc(tools::ToString("shmem: could not create a message of size ", size, ", alignment: ", (alignment != 0) ? std::to_string(alignment) : "default"));
                 }
+                static thread_local tools::RateLimiter rateLimiter(20);
```

Comment:
This changes semantics but may be ok: every thread is now limited to 20 `boost::interprocess::bad_alloc` exceptions per second. If there's a longer time where it doesn't throw it may initially throw at a much higher rate than 20Hz to make up for the "missed throws". This may lead to unintended thrashing though.

Reply:
@rbx Maybe the real question here is how exact you need the rate to be. I guess it's unimportant to reach exactly 20Hz. So how about a simple `this_thread::sleep(50ms)`?

Reply:
It doesn't have to be precise. In fact I just recently changed it from `std::this_thread::sleep_for(50ms)` to rateLimiter, with the hope to avoid some sleeps, but I didn't consider the time calculation in the ctor. The change didn't have any impact on my benchmark performance, so I'm fine with changing it back.

Reply:
Alright. Sounds better than having a `static RateLimiter` here. You're of course right that exception handling is dog slow and thus the sleep might be unnecessary to reach 20Hz. However, the sleep helps to back off and let other threads to their work to free the memory again. (still no fairness, of course)

Reply:
For the allocation method, we can switch back to `std::this_thread::sleep_for(50ms)`. But for the rate limiter in the `RunWrapper` in `FairMQDevice.cxx` precision is more important. There  it is constructed only once per Running state, so the overhead from the constructor is negligible imo.

Reply:
So the 2x faster message allocation is only due to the rate limiter in Message.h? Is that what you see @ironMann? In that case I'd refrain from changing the code in FairMQDevice.cxx.

Reply:
I only inspected the Message case, in the fig above. Though I see gettime in profiles A LOT, like it's everyone's favorite syscall (why isn't it fetched from the vDSO?) 
I'll drop the ratelimit commit, and @rbx please do find a suitable way to handle the bad_alloc throttling.

Reply:
I'd love to use a `clock` that doesn't need a syscall. Do you know of any? I mean, in principle, on x86 `rdtscp` should suffice - if you can determine the correct conversion factor to seconds.

Reply:
apparently it's a configuration of glibc: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=59177
This really should not be a syscall on linux. Will have to test on Centos8